### DISCUSSION TITLE
SETUP WLED

### DISCUSSION THREAD

**Discussion by koenigd79** [photobooth-app/photobooth-app#241] [Howtos and camera specific topics]
Hi,

i installed WLED as discribed but how can i found out the "serial port to the ESP device".

Is this the ip of WLED ??

Daniel

**Comment by mgineer85** [comment]
Hi!
Usually you find it in /dev/serial/by-id ...:
![image](https://github.com/photobooth-app/photobooth-app/assets/3868450/b328979d-1748-4ba3-aa17-9d7c48b94321)

So in this case you would use:
/dev/serial/by-id/usb-1a86_USB2.0-Serial-if00-port0

**Comment by rektosaure** [comment]
I'm curious if I could use something like the Athom Slim (https://www.athom.tech/blank-1/wled-slim-led-strip-controller) with a USB port for control instead of relying on WiFi.

Has anyone had experience with these types of devices?
My initial attempt wasn't successful as the app failed to initialize WLED.

**Reply by mgineer85** [reply]
I use wled on esp and just tested the most recent version. Works fine. If it’s running wled I do not see why it should not work. But I have no explicit experience with that device.

**Reply by mgineer85** [reply]
With wled integration you use usb. You use WiFi only to configure.

**Reply by rektosaure** [reply]
Well not so easy to implement to me.

Failing with :
````WARNING	31.mai.24 09:03:04	failed to init WLED! check device, connection and config. WLED module failed. Please check wiring, device, connection and config.	photobooth.services.baseservice.WledService	connect	71
CRITICAL	31.mai.24 09:03:04	WLED module failed. Please check wiring, device, connection and config.	photobooth.services.baseservice.WledService	device_init	125
INFO	31.mai.24 09:03:04	close port to WLED module	photobooth.services.baseservice.WledService	device_init	121
DEBUG	31.mai.24 09:03:04	wled_detected=False	photobooth.services.baseservice.WledService	device_init	116
CRITICAL	31.mai.24 09:03:04	WLED device did not respond during setup. Check device and connections! no answer from serial WLED device received within timeout	photobooth.services.baseservice.WledService	device_init	114
````

After reaching Athom support, they said usb can only be used to flash firmware and not to control wled and need to rewrite the firmware.
In short, D1 mini looks fine/better/cheaper for small setups like photobooth.

**Reply by mgineer85** [reply]
So you set the serial port correctly? 
please provide full debug log and config.

**Reply by rektosaure** [reply]
Nop I gave up with Athom and used D1 Mini instead.

**Reply by jbhoorasingh** [reply]
Hi, Is there a way to adjust the timing of the lighting and transition? The sequence is a tad bit off on my box.

**Reply by mgineer85** [reply]
You can adjust the wled freely, just keep the preset ids.
Also there are countdown offsets in the app config that allow to compensate countdown and I think it will affect wled also

**Comment by McKaiver** [comment]
I'm also facing some issues.
![20250404_23h37m52s_grim](https://github.com/user-attachments/assets/70b875b1-d4e9-4eb9-9c6a-f4e1f9d018a6)
![20250404_23h43m41s_grim](https://github.com/user-attachments/assets/215bb4ab-5d6b-4691-8e55-18f753c53c56)
so the Port should be configured properly
![grafik](https://github.com/user-attachments/assets/b6c92a11-ff03-4a72-845b-811ac877b58b)
Sync interface is set to:
![grafik](https://github.com/user-attachments/assets/f8d2b57e-90c2-4245-9b61-8aa39fbc8ca5)
![grafik](https://github.com/user-attachments/assets/57a70d32-d391-4516-8dfd-e79644375678)


I connectred the WLED to a USB hub and directly.

**Reply by mgineer85** [reply]
Pls double check your port. Its not correct.
Also use default baud rate.

**Reply by McKaiver** [reply]
well, i checked my installation folder:
![grafik](https://github.com/user-attachments/assets/2c57c309-0e74-46c5-80a3-6e0a936b69ed)
no plugin folder there...

**Reply by McKaiver** [reply]
in the log you can find, what i put into the config and how can i check different than:/dev/serial/by-id

**Reply by McKaiver** [reply]
ok, it works, i copied the port once again and it works no, maybe i copied a space in the end with it?

**Reply by mgineer85** [reply]
The _ was missing

**Reply by McKaiver** [reply]
you are so right, i took your explanation from the second post(because there was the path AND the Portname together).
I compared with my output and just compared the numbers.  because i thought, the text is always the same. 

again learned a lesson, thank you.

**Reply by mgineer85** [reply]
Well, ... I have good news for you:
![grafik](https://github.com/user-attachments/assets/5c0e802f-a7fd-4738-92bd-13defe4c4d97)

Cameras, Serial ports are listed in the next version as files are listed already today, so you just have to pick the right one :)

**Reply by McKaiver** [reply]
ok, without exaggeration, you are great :-D

**Reply by mgineer85** [reply]
Works :)
![Animation](https://github.com/user-attachments/assets/b3a91b3a-56f6-4edf-8996-6fc187cb5db0)
